File: Python/6 Kyu/sum_consecutives.py
def sum_consecutives(s): 
    
    lista = []

    atual = s[0]
    copy = atual
    for x in s[1:]:
        if x== atual:
            copy+=atual
            atual = x
        else:
            lista.append(copy)
            copy = 0 
            atual = x
            copy = x
    lista.append(copy)
    return lista


print(sum_consecutives([1,4,4,4,0,4,3,3,1]))

File: Rust/6 Kyu/sum_consecutives.rs
/// Just to make code nicer
type Number = i32;
type Numbers = Vec<Number>;

/// Sums the numbers that are the same and consecutive.
fn sum_consecutives(numbers: &Numbers) -> Numbers {
    let copia  =  numbers;
    let mut lista = vec![];
    let mut atual  = copia[0];
    let mut copy = atual;
    
    for x  in &copia[1..]{
        if *x == atual{
            
            copy+=atual;
            atual = *x;
        }else{
            lista.push(copy);
            copy = 0;
            atual = *x;
            copy = *x;
            
        }
    }
    lista.push(copy);
    lista